core/src/com/unciv/ui/utils/AskNumberPopup.kt

```diff
@@ -0,0 +1,114 @@
+package com.unciv.ui.utils
+
+import com.badlogic.gdx.graphics.Color
+import com.badlogic.gdx.scenes.scene2d.ui.Button
+import com.badlogic.gdx.scenes.scene2d.ui.Table
+import com.badlogic.gdx.scenes.scene2d.ui.TextField
+
+/** Simple class for showing a prompt for a positive integer to the user
+ * @param screen The previous screen the user was on
+ * @param label A line of text shown to the user
+ * @param icon Icon at the top, should have size 80f
+ * @param defaultText The text that should be in the prompt at the start
+ * @param amountButtons Buttons that when clicked will add/subtract these amounts to the number
+ * @param bounds The bounds in which the number must lie. If one is provided, both must be provided,
+ * but `null` can be used as a 'there is no bounds' value.
+ * @param errorText Text that will be shown when an error is detected
+ * @param actionOnOk Lambda that will be executed after pressing 'OK'.
+ */
+
+class AskNumberPopup(
+    screen: CameraStageBaseScreen,
+    label: String = "Please enter a number",
+    icon: IconCircleGroup = ImageGetter.getImage("OtherIcons/Pencil").apply { this.color = Color.BLACK }.surroundWithCircle(80f),
+    defaultText: String = "",
+    amountButtons: List<Int> = listOf(),
+    bounds: Array<Int?> = arrayOf(null, null), 
+    errorText: String = "Invalid input! Please enter a valid number.",
+    actionOnOk: (input: Int) -> Boolean = { true },
+): Popup(screen) {
```

Comment:
Not building on top of AskTextThingy? interesting.

Reply:
The code inside the `init` function differs on multiple points. The only way I could have built on top of the `AskTextPopup` was if I created a `AskInputPopup` which has a parameter asking for int/string input, which, although reducing code duplication, feels like overcomplication